libraries/fwd/computeFwd/compute_fwd.cpp

```diff
@@ -2248,33 +2267,35 @@ void ComputeFwd::calculateFwd() const
     printf("\nFinished.\n");
 
 out : {
-        //        if (out)
-        //            fclose(out);
-        for (k = 0; k < nspace; k++)
-            if(spaces[k])
-                delete spaces[k];
-        if(mri_head_t)
-            delete mri_head_t;
-        if(meg_head_t)
-            delete meg_head_t;
-        FREE_41(megchs);
-        FREE_41(eegchs);
-        if(megcoils)
-            delete megcoils;
-        if(eegels)
-            delete eegels;
-
-        if(meg_forward)
-            delete meg_forward;
-        if(eeg_forward)
-            delete eeg_forward;
-        if(meg_forward_grad)
-            delete meg_forward_grad;
-        if(eeg_forward_grad)
-            delete eeg_forward_grad;
-
-        if (!res)
-            qCritical("err_print_error();");//err_print_error();
+
+        // TODO: This only temporary until we have the fwd dlibrary refactored. This is only done in order to provide easy testing in test_forward_solution.
+//        //        if (out)
+//        //            fclose(out);
+//        for (k = 0; k < nspace; k++)
+//            if(spaces[k])
+//                delete spaces[k];
+//        if(mri_head_t)
+//            delete mri_head_t;
+//        if(meg_head_t)
+//            delete meg_head_t;
+//        FREE_41(megchs);
+//        FREE_41(eegchs);
+//        if(megcoils)
+//            delete megcoils;
+//        if(eegels)
+//            delete eegels;
+
+//        if(meg_forward)
+//            delete meg_forward;
+//        if(eeg_forward)
+//            delete eeg_forward;
+//        if(meg_forward_grad)
+//            delete meg_forward_grad;
+//        if(eeg_forward_grad)
+//            delete eeg_forward_grad;
+
+//        if (!res)
+//            qCritical("err_print_error();");//err_print_error();
```

Comment:
And why is this commented out?

Reply:
see comment above. We should not delete here since we want to access the data via test_forward_solution after we wrote to file. This is temporary as well.